src/server/req_manager.c

```diff
@@ -1695,15 +1715,37 @@ void
 mgr_sched_unset(struct batch_request *preq)
 {
 	int	  bad_attr = 0;
-	svrattrl *plist;
+	svrattrl *plist, *tmp_plist;
 	int	  rc;
 	pbs_sched *psched = find_scheduler(preq->rq_ind.rq_manager.rq_objname);
 	if (!psched) {
 		req_reject(PBSE_UNKSCHED, 0, preq);
 		return;
 	}
-	plist = (svrattrl *)GET_NEXT(preq->rq_ind.rq_manager.rq_attr);
 
+
+	for (tmp_plist = (svrattrl *)GET_NEXT(preq->rq_ind.rq_manager.rq_attr); tmp_plist; tmp_plist = (struct svrattrl *)GET_NEXT(tmp_plist->al_link)) {
+		if (strcasecmp(tmp_plist->al_name, ATTR_sched_log) == 0 ||
+			strcasecmp(tmp_plist->al_name, ATTR_sched_priv) == 0) {
+			set_scheduler_flag(SCH_ATTRS_CONFIGURE, psched);
+		} else if (strcasecmp(tmp_plist->al_name, ATTR_schediteration) == 0) {
+			if (dflt_scheduler) {
+				svrattrl *t_list;
+				t_list = attrlist_create(tmp_plist->al_name, NULL, 0);
+				if (t_list == NULL) {
+					reply_badattr(-1, bad_attr, tmp_plist, preq);
+				}
+				t_list->al_link.ll_next->ll_struct = NULL;
```

Comment:
where to we free t_list? I think we are leaking this memory

Reply:
Implemented.